src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php

```diff
@@ -125,6 +123,36 @@ public function match(\$rawPathinfo)
         throw \$allow ? new MethodNotAllowedException(array_keys(\$allow)) : new ResourceNotFoundException();
     }
 EOF;
+
+        if ($supportsRedirections) {
+            return <<<'EOF'
+    public function match($pathinfo)
+    {
+        try {
+            return $this->doMatch($pathinfo);
+        } catch (ResourceNotFoundException $e) {
+            if (!in_array($this->context->getMethod(), array('HEAD', 'GET'), true)) {
+                throw $e;
+            }
+
+            try {
+                $pathinfo = '/' !== $pathinfo[-1] ? $pathinfo.'/' : substr($pathinfo, 0, -1);
+                $ret = $this->doMatch($pathinfo);
+
+                return $this->redirect($pathinfo, $ret['_route'] ?? null) + $ret;
```

Comment:
`$ret['_route']` is always set isn't it? So there would be no need for `??`

Reply:
correct, fixed